docs/imagecustomizer/api/configuration.md

```diff
@@ -0,0 +1,147 @@
+# Configuration
+
+Image customizations can be configured using a YAML (or JSON) file.
+
+## Schema Overview
+
+- [Configuration](#configuration)
+  - [Schema Overview](#schema-overview)
+    - [iso \[iso\]](#iso-iso)
+    - [os \[os\]](#os-os)
+    - [pxe \[pxe\]](#pxe-pxe)
+    - [scripts \[scripts\]](#scripts-scripts)
+    - [storage \[storage\]](#storage-storage)
+  - [Operation ordering](#operation-ordering)
+  - [/etc/resolv.conf](#etcresolvconf)
+  - [Replacing packages](#replacing-packages)
+
+### iso [[iso](./iso.md)]
+
+Optionally specifies the configuration for the generated ISO media.
+
+### os [[os](./os.md)]
+
+Contains the configuration options for the OS.
+
+Example:
+
+```yaml
+os:
+  hostname: example-image
+```
+
+### pxe [[pxe](./pxe.md)]
+
+Optionally specifies the PXE-specific configuration for the generated OS
+artifacts.
+
+### scripts [[scripts](./scripts.md)]
+
+Specifies custom scripts to run during the customization process.
+
+### storage [[storage](./storage.md)]
+
+## Operation ordering
+
+1. If partitions were specified in the config, customize the disk partitions.
```

Comment:
There should be empty lines:

- Before and after a heading
- Before and after a multiline codeblock
- Before and after each item in the `Operation ordering` list.

The Markdown file should be human readable. Not just machine readable.

Please, install the `markdownlint` VSCode extension. It will tell you about Markdown formatting errors.